### PR TITLE
fix permutedims dispatch, isapprox for complex numbers

### PR DESCRIPTION
**Pull request by GiggleLiu**
several bug fixes.

There should be some tests. Am I supposed to put tests in `CuArrays.jl` and submit another PR?

### CODE REVIEW

**Comment by maleadt** [comment]
I'm not familiar with this code, @SimonDanisch?
Mind explaining the rationale of your fixes? Better keep the individual fixes confined to separate commits with a line of explanation in the commit message next time.
Also, why not put the tests in the GPUArrays test suite, instead of in https://github.com/JuliaGPU/CuArrays.jl/pull/213?

File: src/linalg.jl [hunk 1]
```diff
@@ -80,7 +80,8 @@ function genperm(I::NTuple{N}, perm::NTuple{N}) where N
     ntuple(d-> (@inbounds return I[perm[d]]), Val(N))
 end
 
-function LinearAlgebra.permutedims!(dest::GPUArray, src::GPUArray, perm::NTuple{N, Integer}) where N
+function LinearAlgebra.permutedims!(dest::GPUArray, src::GPUArray, perm) where N
```
Comment (GiggleLiu) [inline src/linalg.jl]: The interface for `permutedims!` in Base is `permutedims!(dest, src::AbstractArray, perm)`. The original interface will dispatch the call to Base for vector perm unexpectedly. However, vector perm is a valid input for perm in GPUArrays.

File: src/mapreduce.jl [hunk 2]
```diff
@@ -189,6 +189,6 @@ function fast_isapprox(x::Number, y::Number, rtol::Real = Base.rtoldefault(x, y)
     x == y || (isfinite(x) && isfinite(y) && abs(x - y) <= atol + rtol*max(abs(x), abs(y)))
 end
 
-Base.isapprox(A::GPUArray{T1}, B::GPUArray{T2}, rtol::Real = Base.rtoldefault(T1, T2, 0), atol::Real=0) where {T1, T2} = all(fast_isapprox.(A, B, T1(rtol), T1(atol)))
-Base.isapprox(A::AbstractArray{T1}, B::GPUArray{T2}, rtol::Real = Base.rtoldefault(T1, T2, 0), atol::Real=0) where {T1, T2} = all(fast_isapprox.(A, Array(B), T1(rtol), T1(atol)))
-Base.isapprox(A::GPUArray{T1}, B::AbstractArray{T2}, rtol::Real = Base.rtoldefault(T1, T2, 0), atol::Real=0) where {T1, T2} = all(fast_isapprox.(Array(A), B, T1(rtol), T1(atol)))
+Base.isapprox(A::GPUArray{T1}, B::GPUArray{T2}, rtol::Real = Base.rtoldefault(T1, T2, 0), atol::Real=0) where {T1, T2} = all(fast_isapprox.(A, B, T1(rtol)|>real, T1(atol)|>real))
```
Comment (GiggleLiu) [inline src/mapreduce.jl]: `tol` should never be complex numbers.

**Pull request merged**